evennia/utils/tests/test_text2html.py

```diff
@@ -0,0 +1,180 @@
+"""Tests for text2html """
+
+from django.test import TestCase
+from evennia.utils import ansi, text2html
+import mock
+
+
+class TestText2Html(TestCase):
+
+    def test_re_color(self):
+        parser = text2html.HTML_PARSER
+        self.assertEqual("foo", parser.re_color("foo"))
+        self.assertEqual(
+            "<span class=\"color-001\">red</span>foo",
+            parser.re_color(ansi.ANSI_UNHILITE + ansi.ANSI_RED + "red" + ansi.ANSI_NORMAL + "foo"))
+        self.assertEqual(
+            "<span class=\"bgcolor-001\">red</span>foo",
+            parser.re_color(ansi.ANSI_BACK_RED + "red" + ansi.ANSI_NORMAL + "foo"))
+        self.assertEqual(
+            "<span class=\"bgcolor-001\"><span class=\"color-002\">red</span></span>foo",
+            parser.re_color(ansi.ANSI_BACK_RED + ansi.ANSI_UNHILITE + ansi.ANSI_GREEN + "red" + ansi.ANSI_NORMAL + "foo"))
+
+    def test_re_bold(self):
+        parser = text2html.HTML_PARSER
+        self.assertEqual("foo", parser.re_bold("foo"))
+        self.assertEqual(
+            # "a <strong>red</strong>foo",  # TODO: why not?
```

Comment:
If this is really the output then this is obviously wrong behavior - the visual result of this is a bold `redfoo` rather than a bold `red`.

Reply:
Yes. The current result is "**redfoo**" (with the ANSI_NORMAL char seq still present) although I was expecting it to be "**red**foo".

Sounds like a plan!

Can you 2x check each of the TODOs to make sure I am not misinterpreting the intention of the code? I'll use those as citations in the new bug.

Meanwhile, I'll [unitest.skip](https://docs.python.org/3/library/unittest.html#skipping-tests-and-expected-failures) each.

Reply:
#1974 created. Waiting for specific citations.

Reply:
Test is now `unittest.skip()`ed